src/App.tsx

```diff
@@ -1,27 +1,91 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
-import { GoodsList } from './GoodsList';
+import 'bulma';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { LoadingButton } from './components/LoadingButton';
+import { GoodsList } from './components/GoodsList/GoodsList';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+import * as goodsAPI from './api/goods';
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [isInitialized, setIsInitialized] = useState(false);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const [loadedGoodsGroupText, setLoadedGoodsGroupText] = useState('');
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const handleGoodsLoad = useCallback(
+    (getGoodsFromServer: () => Promise<Good[]>) => (
+      async (stopButtonLoading: () => void, goodsGroupText: string) => {
+        setIsLoading(true);
+        setLoadedGoodsGroupText(goodsGroupText);
+
+        try {
+          const goodsFromServer = await getGoodsFromServer();
+
+          setGoods(goodsFromServer);
+          setIsInitialized(true);
```

Comment:
why do you need this state? it seem to be similar to `!hasLoadingError && !isLoading` result

Reply:
I need to use `setIsInitialized` to handle the situation where no goods are received from the server, and display a corresponding message. So I change it once when the user clicks on a button for the first time.

Without `setIsInitialized` a "_no goods_" message would be displayed even BEFORE the user has made his first request (like on the picture), which is not the desired behavior.
![image](https://user-images.githubusercontent.com/50755505/230123536-9e311199-58e8-4bf9-86ee-3b0eb635f5e2.png)